### PR TITLE
update ginkgo, gomega and other unreasonable code to fix the CI

### PR DESCRIPTION
**Pull request by Thor-wl**
Fix the CI failure
Signed-off-by: Thor-wl <13164644535@163.com>

### CODE REVIEW

File: hack/run-e2e-kind.sh [hunk 1]
```diff
@@ -133,35 +133,35 @@ GO111MODULE=off go get github.com/onsi/ginkgo/ginkgo
 case ${E2E_TYPE} in
 "ALL")
     echo "Running e2e..."
-    KUBECONFIG=${KUBECONFIG} ginkgo -r --nodes=4 --compilers=4 --randomizeAllSpecs --randomizeSuites --failOnPending --cover --trace --race --slowSpecThreshold=30 --progress ./test/e2e/jobp/
-    KUBECONFIG=${KUBECONFIG} ginkgo -r --slowSpecThreshold=30 --progress ./test/e2e/jobseq/
-    KUBECONFIG=${KUBECONFIG} ginkgo -r --slowSpecThreshold=30 --progress ./test/e2e/schedulingbase/
-    KUBECONFIG=${KUBECONFIG} ginkgo -r --slowSpecThreshold=30 --progress ./test/e2e/schedulingaction/
-    KUBECONFIG=${KUBECONFIG} ginkgo -r --slowSpecThreshold=30 --progress ./test/e2e/vcctl/
+    KUBECONFIG=${KUBECONFIG} ginkgo -r --nodes=4 --compilers=4 --randomizeAllSpecs --randomizeSuites --failOnPending --cover --trace --race -slow-spec-threshold='30s' --progress ./test/e2e/jobp/
```
Comment (hwdef) [inline hack/run-e2e-kind.sh:136]:
```suggestion
    KUBECONFIG=${KUBECONFIG} ginkgo -r --nodes=4 --compilers=4 --randomizeAllSpecs --randomizeSuites --failOnPending --cover --trace --race --slow-spec-threshold='30s' --progress ./test/e2e/jobp/
```

**Comment by william-wang** [comment]
make this pr only for "ginkgo and gomega".  the change for "cache", "job_info" should be submitted with a new seperated pr.

**Comment by Thor-wl** [comment]
> make this pr only for "ginkgo and gomega". the change for "cache", "job_info" should be submitted with a new seperated pr.

Sorry for not state it clear. This PR is to fix the CI failure. It consists of dependency update and code fix for code verification failure.

File: pkg/scheduler/cache/interface.go [hunk 2]
```diff
@@ -80,7 +80,7 @@ type VolumeBinder interface {
 
 //Binder interface for binding task and hostname
 type Binder interface {
-	Bind(kubeClient *kubernetes.Clientset, tasks []*api.TaskInfo) (error, []*api.TaskInfo)
```
Comment (william-wang) [inline pkg/scheduler/cache/interface.go:83]: Need to consider compatibility. The change will impact the bind implementation in user side.

File: pkg/scheduler/cache/cache.go [hunk 3]
```diff
@@ -152,14 +152,14 @@ func (db *defaultBinder) Bind(kubeClient *kubernetes.Clientset, tasks []*schedul
...
-		return fmt.Errorf("failed to bind pods"), errTasks
+		return errTasks, fmt.Errorf("failed to bind pods")
 	}
 
 	return nil, nil
 }
 
-func NewBinder() *defaultBinder {
-	return &defaultBinder{}
+func NewBinder() *DefaultBinder {
```
Comment (william-wang) [inline pkg/scheduler/cache/cache.go:161]: This changes the scope of "DefaultBinder".

File: pkg/scheduler/cache/interface.go (hunk 2 above)
Comment (Thor-wl) [reply pkg/scheduler/cache/interface.go:83]: If not fix like this, the code verify will not be passed.

File: pkg/scheduler/cache/cache.go (hunk 3 above)
Comment (Thor-wl) [reply pkg/scheduler/cache/cache.go:161]: > This changes the scope of "DefaultBinder".

Same as above.

**Review by william-wang — APPROVED** [review]
/lgtm

File: pkg/scheduler/cache/cache.go (hunk 3 above)
Comment (k82cn) [reply pkg/scheduler/cache/cache.go:161]: why change the scope ?

**Comment by k82cn** [comment]
please seperate such kind of pr into at least two part: fix in pkg and vendor update.

**Pull request merged**